warehouse/packaging/views.py

```diff
@@ -74,11 +77,22 @@ def release_detail(release, request):
     all_releases = (
         request.db.query(Release)
                   .filter(Release.project == project)
-                  .with_entities(Release.version, Release.created)
+                  .with_entities(
+                        Release.version,
+                        Release.is_prerelease,
+                        Release.created)
                   .order_by(Release._pypi_ordering.desc())
                   .all()
     )
 
+    # Get the latest non-prerelease version of this Project, or the latest
+    # of any version if there are no non-prerelease versions.
```

Comment:
This comment took me a second to wrap my head around, perhaps it would be more clear if it said:
```
Get the latest non-prerelease of this Project, or the latest release if all
releases are prereleases.
```